### PR TITLE
Implement Issue #4 & #14

### PR DESCRIPTION
**Pull request by dannicolici**
- add support for 'as' in the 'given' macro

### CODE REVIEW

File: test/eggplant/example_specification_test.clj [hunk 1]
```diff
@@ -5,28 +5,41 @@
...
   (testing "Test with Thread Macro given"
-    (given 2 test-data
+    (given 2 as test-data
            (then-we-expect 4
                            (when-we-process * 2 test-data)))))
 
+(defspec example-specification-vector-as-data
+         (testing "Test with Thread Macro given"
+           (given [1 2 3 4] as test-data
+                  (then-we-expect 4
```
Comment (perkss) [inline test/eggplant/example_specification_test.clj]: Formatting here is quite different is the only thing. Real nice though

File: src/eggplant/core.clj [hunk 2]
```diff
@@ -1,9 +1,16 @@
 (ns eggplant.core
   (:require [clojure.test :refer :all]))
 
+(defn multi-args? [data]
```
Comment (perkss) [inline src/eggplant/core.clj]: could make this private?

Comment (dannicolici) [reply src/eggplant/core.clj]: done

File: test/eggplant/example_specification_test.clj (hunk 1 above)
Comment (dannicolici) [reply test/eggplant/example_specification_test.clj]: done

**Review by perkss — APPROVED** [review]
LGTM

**Pull request merged**